src/typegraph/graph/src/visitor.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::{interlude::*, path::RelativePath, TypeNode as _};
+use crate::{Edge, EdgeKind, Type, TypeNodeExt as _};
+
+pub enum VisitNext {
+    Children,
+    Siblings,
+    Stop,
+}
+
+pub struct VisitNode {
+    pub ty: Type,
+    pub path: Vec<Edge>,
+}
+
+pub fn traverse_types<A, V, E: From<color_eyre::eyre::Error>>(
+    root: Type,
+    accumulator: A,
+    visit_fn: V,
+) -> Result<A, E>
+where
+    V: Fn(&VisitNode, &mut A) -> Result<VisitNext, E>,
+{
+    let mut path = Vec::new();
+    traverse_types_with_path(root, &mut path, accumulator, &visit_fn)
+        .map(|output| output.accumulator)
+}
+
+struct TraverseOutput<A> {
+    accumulator: A,
+    stop: bool,
+}
+
+fn visit<A, V, E: From<color_eyre::eyre::Error>>(
+    node: Type,
+    path: &mut Vec<Edge>,
+    accumulator: &mut A,
+    visit_fn: V,
```

Comment:
I am seeing that type V is the only point accumulator is used, why not rm it from the args?

Reply:
(...) to get the initial value...